components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -98,6 +99,14 @@ class DownloadsFeature(
         }
     }
```

Comment:
nit: While we are here, this code seems to not need `let` and could be written as concise as your `onPermissionDenied()` code.

Reply:
@pocmo hm, but we need to capture a non-null session. So, this won't work:
```kotlin
 activeSession?.download?.consume {
   onDownload(activeSession, it)
  }
```
because `activeSession` is a mutable property that could have been changed by this time :). So, I think let is the best way here?

Reply:
Oh, yeah, makes sense!